lib/utils.js

```diff
@@ -392,6 +392,29 @@ var Utils = module.exports = {
 
   validateParameter: function(value, expectation, options) {
     return ParameterValidator.check(value, expectation, options);
+  },
+
+  formatReferences: function (obj) {
+    var references = {};
+
+    if (!obj || !_.isPlainObject(obj) || !obj.references) {
+      return obj;
+    }
+
+    if (_.isPlainObject(obj.references)) {
+      references = obj.references;
+    } else {
```

Comment:
You do not need this case: if the obj.references is already a plain object, it'll stay that way.